src/components/displaybook.js

```diff
@@ -1,13 +1,76 @@
-/* eslint-disable react/prop-types */
-export default function Book({ book }) {
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getBookItems, removeBook } from '../redux/books/booksSlice';
+
+const Book = () => {
+  const dispatch = useDispatch();
+  const { books, isLoading, error } = useSelector((state) => state.books);
+
+  useEffect(() => {
+    dispatch(getBookItems());
+  }, [dispatch]);
+
+  const renderBook = (itemId, book) => {
+    const { category, title, author } = book;
+    return (
+      <div key={itemId} className="containerRender" style={{ display: 'flex' }}>
+        <div className="renderLeft">
+          <p>{category}</p>
+          <h3>{title}</h3>
+          <p>{author}</p>
+          <div className="leftButton">
+            <button className="leftButton" type="button">
+              Comment
+            </button>
+            <button
+              className="leftButton"
+              type="button"
+              onClick={() => dispatch(removeBook(itemId))}
+            >
+              Remove
+            </button>
+            <button className="leftButton" type="button">
+              Edit
+            </button>
+          </div>
+        </div>
+        <div className="renderRight">
+          <div className="middle">
+            <p className="Amid">Progress</p>
+            <div className="midRight">
+              <h4>60%</h4>
+              <h4>Completed</h4>
+            </div>
+          </div>
+          <div className="deepRight">
+            <p>Current Chapter</p>
+            <p>Chapter 15</p>
+            <button className="blue" type="button">
+              UPDATE PROGRESS
+            </button>
+          </div>
+        </div>
+      </div>
+    );
+  };
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (error) {
+    return <p>Request failed</p>;
+  }
+
+  if (!books) {
+    return null;
+  }
+
   return (
-    <>
-      <h1>{ book.bookname }</h1>
-      <p>
-        <strong>Auther: </strong>
-        { book.auther }
-      </p>
-      <button type="button">Delete</button>
-    </>
+    <div>
+      {Object.entries(books).map(([itemId, book]) => renderBook(itemId, book[0]))}
+    </div>
```

Comment:
- [Optional] Dear @ethred, I notice the full API rerenders whenever we add a new book or remove a book. It's costly to hit a remote API and let the DOM refresh while doing so. I should suggest removing the element/book from the DOM and sending the request to the API without rendering the DOM. The same thing wen adding a new book.